main.js

```diff
@@ -0,0 +1,158 @@
+// images links
+const images = [
+  "./images/image5.jpg",
+  "./images/image1.jpg",
+  "./images/image2.jpg",
+  "./images/image3.jpg",
+  "./images/image4.jpg",
+  "./images/image5.jpg",
+  "./images/image1.jpg",
+];
+
+// selectors
+const slides = document.querySelector(".slide");
+const nextBtn = document.querySelector(".slider__btn--right");
+const previousBtn = document.querySelector(".slider__btn--left");
+const navigationBar = document.querySelector(".navigation__bar");
+
+// class declaration
+class ImageSlider {
+  constructor(slide, images, document, indicatorsContainer) {
+    this.slide = slide;
+    this.images = images;
+    this.document = document;
+    this.isTransition = false;
+    this.indicatorsContainer = indicatorsContainer;
+    this.slideIndex = 1;
+  }
+
+  //populates images to dom
+  generateImages() {
+    this.images.map((v, i) => {
```

Comment:
`this.images.map((img, ind) => {`
will be clear in understanding the code. Avoid single letter variables

Reply:
resolved